Adaptors/S3/src/ObjectStorage.cs

```diff
@@ -169,17 +168,36 @@ await taskList.WhenAll()
                                       {
                                         Key = $"{objectStorageName_}{key}_count",
                                       },
-                                    })
-                            .ToList();
-    var multiObjectDeleteRequest = new DeleteObjectsRequest
-                                   {
-                                     BucketName = bucketName_,
-                                     Objects    = keyList,
-                                   };
+                                    });
 
-    var deleteObjectsResponse = await s3Client_.DeleteObjectsAsync(multiObjectDeleteRequest)
-                                               .ConfigureAwait(false);
-    return deleteObjectsResponse.DeletedObjects.Count == valuesCount + 1;
+    var deletedItem = 0;
+    try
+    {
+      await keyList.Chunk(MaxAllowedKeysPerDelete)
+                   .Select(async chunkedKeyList =>
+                           {
+                             var multiObjectDeleteRequest = new DeleteObjectsRequest
+                                                            {
+                                                              BucketName = bucketName_,
+                                                              Objects    = null,
+                                                            };
+                             multiObjectDeleteRequest.Objects = chunkedKeyList.ToList();
```

Comment:
maybe you should be able to do something similar to:
```suggestion
                                                              Objects    =  {
                                                                  chunkedKeyList
                                                              },
                                                            };
```

Reply:
it is already the case... now...